src/java.base/share/classes/sun/security/ssl/JsseJce.java

```diff
@@ -393,10 +419,38 @@ private static class EcAvailability {
         // Is EC crypto available?
         private static final boolean isAvailable;
 
+        /**
+         * Checks if a particular signature algorithm is available.
+         *
+         * @param algorithm the algorithm we will attempt to instantiate to check if it is available
+         * @return true if the signature algorithm is found, false otherwise
+         */
+        private static boolean isSignatureAlgorithmAvailable(String algorithm) {
+            try {
+                // Attempt to create a Cipher instance with the specified algorithm.
+                Signature.getInstance(algorithm);
```

Comment:
I think this should use `JsseJce.getSignature()` like line 399 did.

Reply:
Updated to make use of `JsseJce.getSignature()` instead of `Signature.getInstance()`